src/officehours_api/backends/backend_base.py

```diff
@@ -0,0 +1,38 @@
+from abc import ABC, abstractclassmethod, abstractmethod
+from typing import Optional
+
+from django.contrib.auth.models import User
+
+from officehours_api.backends.types import BackendDict, IMPLEMENTED_BACKEND_NAME
+
+
+class BackendBase(ABC):
+
+    name: IMPLEMENTED_BACKEND_NAME
+    friendly_name: str = 'Backend'
+    enabled: bool = True
+    docs_url: Optional[str] = None
+    profile_url: Optional[str] = None
+    telephone_num: Optional[str] = None
+    intl_telephone_url: Optional[str] = None
+
+    @classmethod
+    def get_public_data(cls) -> BackendDict:
+        return {
+            'name': cls.name,
+            'friendly_name': cls.friendly_name,
+            'enabled': cls.enabled,
+            'docs_url': cls.docs_url,
+            'profile_url': cls.profile_url,
+            'telephone_num': cls.telephone_num,
+            'intl_telephone_url': cls.intl_telephone_url
+        }
+    
+    @classmethod
+    @abstractclassmethod
+    def is_authorized(cls, user: User) -> bool:
+        pass
+
+    @abstractmethod
+    def save_user_meeting(self, backend_metadata: dict, assignee: User):
+        pass
```

Comment:
@jlost, this isn't the highest priority issue, but I was curious how difficult it would be. When you get a chance, let me know what you think. I think there might be a way to enforce the class variables, but get_public_data would fail anyway if not all those were defined. We could remove the defaults, but I read somewhere those were recommended when possible.

Maybe relevant reading: https://stackoverflow.com/questions/22046369/enforcing-class-variables-in-a-subclass